Suggested Change in NavigationRail For Longer Text in NavigationRailItem

AnEx07 opened the pull request:
#991 I faced this issue today and here is my solution...
code -> 
```



from kivy.factory import Factory
from kivy.lang import Builder
from kivy.utils import *
from kivymd.app import MDApp

KV = '''
#:import get_color_from_hex kivy.utils.get_color_from_hex
#:import Transition kivy.uix.screenmanager.SwapTransition

<MyTile@SmartTileWithStar>
    size_hint_y: None
    height: "240dp"


MDBoxLayout:
    orientation: "vertical"
    # md_bg_color: [1,1,1,1]

    MDToolbar:
        title: "MDNavigationRail"
        md_bg_color: rail.md_bg_color
        left_action_items: [["menu", lambda x: app.rail_open()]]

    MDBoxLayout:

        MDNavigationRail:
            id: rail
            md_bg_color: get_color_from_hex("#344954")
            color_normal: get_color_from_hex("#718089")
            color_active: get_color_from_hex("#f3ab44")
            use_resizeable: True
            use_hover_behavior: True
            # hover_bg: 0, 0, 0, .2
            visible: "Persistent"
            normal_width: dp(75)
            # open_width: dp(250)
            

            MDNavigationRailItem:
                id: rootItem
                icon: "timer"
                text: "Record Time"
                visible: "Selected"
                
            

            MDNavigationRailItem:
                icon: "calendar-week"
                text: "Weekly Time Entry"


            MDNavigationRailItem:
                icon: "projector-screen"
                text: "Project Manegement"

            MDNavigationRailItem:
                icon: "exit-run"
                text: "Apply Leave"


 
        ScreenManager:
            id: sm
            transition: Transition()
            
            
            
'''


class Test(MDApp):
    def build(self):
        # self.theme_cls.primary_color = get_color_from_hex("#344954")
        # self.theme_cls.theme_style = "Dark"
        return Builder.load_string(KV)

    def rail_open(self):
        if self.root.ids.rail.rail_state == "open":
            self.root.ids.rail.rail_state = "close"
        else:
            self.root.ids.rail.rail_state = "open"




    def on_start(self):
        self.root.ids.rail.item_switch(self.root.ids.rootItem)
        return super().on_start()


Test().run()
    


```
outputs ->
![Screenshot from 2021-09-17 01-58-30](https://user-images.githubusercontent.com/79507388/133681855-7ac4c5d8-06bb-4641-966b-93cd7bc22dcf.png)
![Screenshot from 2021-09-17 01-57-21](https://user-images.githubusercontent.com/79507388/133682084-10526bb6-9000-4d8c-8562-d377ff0b0445.png)

![Screenshot from 2021-09-17 01-58-21](https://user-images.githubusercontent.com/79507388/133681863-ae239939-c225-41c5-9517-0235349167b8.png)
![Screenshot from 2021-09-17 01-57-21](https://user
![Screenshot from 2021-09-16 22-25-08](https://user-images.githubusercontent.com/79507388/133681880-9a38a0c2-4403-4464-bb75-e825cdf79f38.png)
![Screenshot from 2021-09-16 22-22-01](https://user-images.githubusercontent.com/79507388/133681958-8c31ad50-efd0-4ecc-975d-22d6faba5497.png)

pakal committed:
Fix wrong "BoxLayout" occurrence in MD colors sample program (#1079)

AnEx07 committed:
Target to fix longer text issue in MDNavigationRailItem.

1. Adjust BaseNavigationRailBoxItem padding
2. add condintion for MDNavigationRailItem.spacing for correctly position its text
3. MDNavigationRailItem.MDIcon.size now adjust acording to MDNavigationRail.normal_width
4. MDNavigationRailItem.MDIcon pos_hint to center to adjust position with MDNavigationRail.width
5. add condition for MDNavigationRailItem.MDLabel.halign it halign to center while it's orientation is verticale and to left while in horizontal which is in MDNavigationRail.rail_state == "open"
6. adjust spacing in MDNavigationRail.MDList

AnEx07 committed:
Target to fix longer text issue in MDNavigationRailItem in py file.

1. added attr: MDNavigationRailItem.normal_width a NumericProperty default to dp(72) it bind to MDNavigationRail.normal_width
2. commented MDNavigationRailItem.set_width as now width change acording to MDNavigationRailItem.open_width and use size_hint while MDNavigationRailItem.use_resizeable == False
3. insted of seting size now MDNavigationRailItem.set_width only change its width acording to MDNavigationRailItem.open_width though it never called as of now
4. added attr: MDNavigationRail.normal_width a NumericProperty defaults to dp(72) it used for setting width of MDNavigationRail in close state
5. added attr: MDNavigationRail.open_width a NumericProperty defaults to dp(72)*4 it used for setting width of MDNavigationRail in open state
6. changed in MDNavigationRail.open, now using MDNavigationRail.open_width insted of self.width*4 for width for Animation
7. changed in MDNavigationRail.close, now using MDNavigationRail.normal_width insted of self.width/4 for width for Animation
8. adjusted value in MDNavigationRail.on_use_resizeable now MDNavigationRail.ids.box.spacing = "20dp" insted of "24dp"
9. changed in MDNavigationRail.set_width now using MDNavigationRail.normal_width insted of dp(72)

HeaTTheatR commented:
@AnEx07 
Thank you for helping the development of the KiwiMD library.

- But I have some small remarks: your `PR` does not contain an explanation from which people could understand why you made these changes.
- The commits of your changes are not informative. The commit should contain a title and description of the changes.
- Please note that we have a template by which you can make a description of your changes - https://github.com/kivymd/.github/blob/main/.github/pull_request_template.md

HeaTTheatR commented on kivymd/uix/navigationrail/navigationrail.kv:
```diff
@@ -55,36 +55,42 @@
         id: btn_icon
+        # md_bg_color: [0,0.5,0,0.5]
+        size_hint: None,None
+        size: [root.normal_width-root.padding[0]-root.padding[2],self.font_size]
```
There must be a space after the comma (use the PEP8 standard).

HeaTTheatR commented on kivymd/uix/navigationrail/navigationrail.kv:
```diff
@@ -55,36 +55,42 @@
             root.theme_cls.text_color \
             if not root._color_normal else root._color_normal
-        pos_hint: {"center_y": .5}
+        pos_hint: {"center_x": .5,"center_y": .5}
```
There must be a space after the comma (use the PEP8 standard).

HeaTTheatR commented on kivymd/uix/navigationrail/navigationrail.py:
```diff
@@ -554,6 +556,14 @@ def on_start(self):
+    open_width = NumericProperty(dp(72)*4)
+    # width used for open state
+    # also used to set railitems width
+
```
Use a generally accepted standard for describing parameters for `MD` classes:

```python
    normal_width = NumericProperty(dp(72))
    """
    Width used for close state also used for rail items icon width.

    :attr:`normal_width` is an :class:`~kivy.properties.NumericProperty`
    and defaults to `dp(72)`.
    """
```

AnEx07 commented:
should i change those and redo pull request or what?

AnEx07 committed:
Change for PEP8 standard in navigationrail.kv

1. just format the file for PEP8

AnEx07 committed:
Formating and Describing changes in naviationrail.py

1. format as per PEP8
2. Docmenting added property as per kivymd

HeaTTheatR commented:
@AnEx07 
The commits of your changes are not informative. The commit should contain a title and description of the changes.
https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message

HeaTTheatR commented:
@AnEx07 Check out the Git documentation: what the commits to the changed files should look like.

pakal committed:
Fix typo in gridlayout.py (#1086)

"hight" and "width" were mixed in adaptive_width equivalence doc

AnEx07 committed:
Adding handeler for newly added property in navigationrail.py

1. now width change on runtime  with normal_width for "close" state and open_width for "open" state
2. MDNavigationRail.normal_width  bind with MDNavigationRailItem.normal_width which change width of the MDIcon class and place icon in center of  MDNavigationRail.width
3. MDNavigationRail.open_width  bind with MDNavigationRailItem.width which change width of hole MDNavigationRailItem class only when MDNavigationRail.use_resizeable = True which make MDNavigationRailItem.size_hint = (None, None)

AnEx07 committed:
Merge branch 'kivymd:master' into master

AnEx07 commented:
> @AnEx07
> The commits of your changes are not informative. The commit should contain a title and description of the changes.
> https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message

@HeaTTheatR  I  have changed those commits messages can you please review those?

AnEx07 commented:
result of new changes ->

https://user-images.githubusercontent.com/79507388/133854745-3e573824-160a-454c-b801-2814a1d330e9.mp4

https://user-images.githubusercontent.com/79507388/133854754-5bff1a08-51ec-4f2c-85e2-d71a4d9c0e32.mp4

HeaTTheatR commented:
@AnEx07 The point is that the behavior that you demonstrate in the recent videos is not inherent in the `NavigationRail` component. Check out the spec for this component - https://material.io/components/navigation-rail

AnEx07 commented:
Oh I got it...
sorry for the wrong suggestion!

AnEx07 commented:
though my project need this 😅
